src/main/java/com/shrralis/ssdemo1/security/AuthProvider.java

```diff
@@ -73,8 +76,9 @@ private UserDetails retrieveUserDetails(String loginOrEmail) {
 			throw new InternalAuthenticationServiceException(
 					"UserDetailsService returned null, which is an interface contract violation");
 		} else if (userDetails instanceof AuthorizedUser
-				&& ((AuthorizedUser) userDetails).getFailedAuthCount() == MAX_FAILED_AUTH_VALUE) {
-			throw new TooManyNonExpiredRecoveryTokensException(loginOrEmail);
+				&& ((AuthorizedUser) userDetails).getFailedAuthCount() == MAX_FAILED_AUTH_VALUE
```

Comment:
I would use >=

Reply:
I already checked all scenarios for this but O.K., this have some sense